File: java-hfsframework-primefaces/hfsframework-base/src/main/java/br/com/hfsframework/security/model/PaginaVO.java
/**
 * <p><b>HFS Framework</b></p>
 * @author Henrique Figueiredo de Souza
 * @version 1.0
 * @since 2017
 */
package br.com.hfsframework.security.model;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;

/**
 * The Class PaginaVO.
 */
public class PaginaVO implements Serializable {

	/** The Constant serialVersionUID. */
	private static final long serialVersionUID = 1L;

	/** The id. */
	private Long id;

	/** The managed bean. */
	private String managedBean;

	/** The url. */
	private String url;

	/** The funcionalidades. */
	private List<FuncionalidadeVO> funcionalidades;

	/** The perfils. */
	private List<PerfilVO> perfils;

	/**
	 * Instantiates a new pagina VO.
	 */
	public PaginaVO() {
		this.funcionalidades = new ArrayList<FuncionalidadeVO>();
		this.perfils = new ArrayList<PerfilVO>();
		limpar();
	}
	
	/**
	 * Limpar.
	 */
	public void limpar() {
		this.id = null;
		this.managedBean = null;
		this.url = null;
		this.funcionalidades.clear();
		this.perfils.clear();
	}



	/**
	 * Gets the id.
	 *
	 * @return the id
	 */
	public Long getId() {
		return this.id;
	}

	/**
	 * Sets the id.
	 *
	 * @param id
	 *            the new id
	 */
	public void setId(Long id) {
		this.id = id;
	}

	/**
	 * Gets the managed bean.
	 *
	 * @return the managed bean
	 */
	public String getManagedBean() {
		return this.managedBean;
	}

	/**
	 * Sets the managed bean.
	 *
	 * @param managedBean
	 *            the new managed bean
	 */
	public void setManagedBean(String managedBean) {
		this.managedBean = managedBean;
	}

	/**
	 * Gets the url.
	 *
	 * @return the url
	 */
	public String getUrl() {
		return this.url;
	}

	/**
	 * Sets the url.
	 *
	 * @param url
	 *            the new url
	 */
	public void setUrl(String url) {
		this.url = url;
	}

	/**
	 * Gets the funcionalidades.
	 *
	 * @return the funcionalidades
	 */
	public List<FuncionalidadeVO> getFuncionalidades() {
		return this.funcionalidades;
	}

	/**
	 * Sets the funcionalidades.
	 *
	 * @param funcionalidades
	 *            the new funcionalidades
	 */
	public void setFuncionalidades(List<FuncionalidadeVO> funcionalidades) {
		this.funcionalidades = funcionalidades;
	}

	/**
	 * Adiciona o funcionalidades.
	 *
	 * @param funcionalidades
	 *            the funcionalidades
	 * @return the funcionalidade VO
	 */
	public FuncionalidadeVO addFuncionalidades(FuncionalidadeVO funcionalidades) {
		getFuncionalidades().add(funcionalidades);
		funcionalidades.setPaginaInicial(this);

		return funcionalidades;
	}

	/**
	 * Remove o funcionalidades.
	 *
	 * @param funcionalidades
	 *            the funcionalidades
	 * @return the funcionalidade VO
	 */
	public FuncionalidadeVO removeFuncionalidades(FuncionalidadeVO funcionalidades) {
		getFuncionalidades().remove(funcionalidades);
		funcionalidades.setPaginaInicial(null);

		return funcionalidades;
	}

	/**
	 * Gets the perfils.
	 *
	 * @return the perfils
	 */
	public List<PerfilVO> getPerfils() {
		return this.perfils;
	}

	/**
	 * Sets the perfils.
	 *
	 * @param perfils
	 *            the new perfils
	 */
	public void setPerfils(List<PerfilVO> perfils) {
		this.perfils = perfils;
	}
	

	/* (non-Javadoc)
	 * @see java.lang.Object#hashCode()
	 */
	@Override
	public int hashCode() {
		final int prime = 31;
		int result = 1;
		result = prime * result + ((id == null) ? 0 : id.hashCode());
		return result;
	}

	/* (non-Javadoc)
	 * @see java.lang.Object#equals(java.lang.Object)
	 */
	@Override
	public boolean equals(Object obj) {
		if (this == obj)
			return true;
		if (obj == null)
			return false;
		if (getClass() != obj.getClass())
			return false;
		PaginaVO other = (PaginaVO) obj;
		if (id == null) {
			if (other.id != null)
				return false;
		} else if (!id.equals(other.id))
			return false;
		return true;
	}

	/* (non-Javadoc)
	 * @see java.lang.Object#toString()
	 */
	@Override
	public String toString() {
		return this.url;
	}

	/**
	 * Gets the perfis pagina.
	 *
	 * @return the perfis pagina
	 */
	public String getPerfisPagina() {
		String ret = "";
		for (PerfilVO item : getPerfils()) {
			ret = ret.concat(item.getDescricao()).concat(", ");
		}
		if (ret != "") {
			ret = ret.substring(0, ret.length() - 2);
		}
		return ret;
	}
	

}

File: csharp-hfsframework/hfsframework-base/hfsframework.security.model/PaginaVO.cs
using System;
using System.Collections.Generic;

namespace hfsframework.security.model
{
    [Serializable]
    public class PaginaVO
    {
	    public long Id { get; set; }	    
	    public string ManagedBean { get; set; }
        public string Url { get; set; }
        public List<FuncionalidadeVO> Funcionalidades { get; set; }
        public List<PerfilVO> Perfils { get; set; }

        public PaginaVO()
        {
            this.Funcionalidades = new List<FuncionalidadeVO>();
            this.Perfils = new List<PerfilVO>();
            limpar();
        }

        public void limpar()
        {
            this.Id = -1;
            this.ManagedBean = null;
            this.Url = null;
            this.Funcionalidades.Clear();
            this.Perfils.Clear();
        }

        public FuncionalidadeVO AddFuncionalidades(FuncionalidadeVO funcionalidades)
        {
            this.Funcionalidades.Add(funcionalidades);
            funcionalidades.PaginaInicial = this;

            return funcionalidades;
        }

        public FuncionalidadeVO removeFuncionalidades(FuncionalidadeVO funcionalidades)
        {
            this.Funcionalidades.Remove(funcionalidades);
            funcionalidades.PaginaInicial = null;

            return funcionalidades;
        }

        public string GetPerfisPagina()
        {
            string ret = "";
            foreach (PerfilVO item in Perfils)
            {
                ret += ret + item.Descricao + ", ";
            }
            if (ret != "")
            {
                ret = ret.Substring(0, ret.Length - 2);
            }
            return ret;
        }

        public override bool Equals(object obj)
        {
            if (obj == null)
                return false;
            PaginaVO outro = (PaginaVO)obj;
            if (outro == null)
                return false;
            if (Id.Equals(outro.Id))
                return true;
            return false;
        }

        public override int GetHashCode()
        {
            return Id.GetHashCode();
        }

        public override string ToString()
        {
            return Url;
        }

    }
}
